source/Coop/NetImpl/PlatformAPI.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Coop.NetImpl.Steam;
+using Epic.OnlineServices;
+using Galaxy.Api;
+using Network.Infrastructure;
+using NLog;
+using Steamworks;
+using System;
+using System.Text;
+using TaleWorlds.Library;
+using TaleWorlds.PlatformService;
+using TaleWorlds.PlatformService.Steam;
+using TaleWorlds.PlayerServices;
+
+namespace Coop.NetImpl
+{
+    //This is just here to make our lives easier when using TaleWorlds PlatformServices and extend them when needed
+    public class PlatformAPI
+    {
+        private IFriendListService friendListService;
+        private IPlatformServices platformServices;
+        private static readonly NLog.Logger Logger = LogManager.GetCurrentClassLogger();
+
+        /*
+         * This constructor should basically only be used for testing. 
+        * Since the services are retrieved during runtime from TaleWorlds' PlatformService (other than in unit tests),
+        * use PlatformAPI() to make a PlatformAPI with the retrieved services from the current platform.
+        */
+        public PlatformAPI(IFriendListService friendListService, IPlatformServices platformServices)
+        {
+            this.friendListService = friendListService;
+            this.platformServices = platformServices;
+        }
+
+        public PlatformAPI()
+        {
+            platformServices = PlatformServices.Instance;
+            platformServices.Initialize(new IFriendListService[0]);
+            friendListService = platformServices.GetFriendListServices()[0];
```

Comment:
Thinking about it, you might want to catch the case where no friend list service is available. We cannot really control that on our end. Right now that would lead to a nullptr access

Reply:
How do you want to handle it? Log it as an error and maybe throw an exception since the PlatformAPI is essentially broken without the friend list service? Trying to think of how this would affect our use cases for it. Maybe attempt to construct the PlatformAPI when connecting to a session and abort the connection attempt if it throws the exception?

Reply:
On second thought, the friends list service isn't required to get the UserId, only the PlatformService instance is... but then again, a given platform's UserId may not technically be unique between platforms, so maybe we avoid it anyway? (PlayerId, not UserId is TaleWorlds' ID that prefixes some stuff to differentiate between platforms and maybe some other stuff)